java/src/main/java/ai/rapids/cudf/HostColumnVector.java

```diff
@@ -441,6 +450,68 @@ public static HostColumnVector timestampNanoSecondsFromLongs(long... values) {
     return build(DType.TIMESTAMP_NANOSECONDS, values.length, (b) -> b.appendArray(values));
   }
 
+  /**
+   * Create a new decimal vector from unscaled values (int array) and scale.
+   * The created vector is of type DType.DECIMAL32, whose max precision is 9.
+   * Compared with scale of [[java.math.BigDecimal]], the scale here represents the opposite meaning.
+   */
+  public static HostColumnVector decimalFromInts(int scale, int... values) {
+    return build(DType.create(DType.DTypeEnum.DECIMAL32, scale), values.length, (b) -> b.appendUnscaledDecimalArray(values));
+  }
+
+  /**
+   * Create a new decimal vector from unscaled values (long array) and scale.
+   * The created vector is of type DType.DECIMAL64, whose max precision is 18.
+   * Compared with scale of [[java.math.BigDecimal]], the scale here represents the opposite meaning.
+   */
+  public static HostColumnVector decimalFromLongs(int scale, long... values) {
+    return build(DType.create(DType.DTypeEnum.DECIMAL64, scale), values.length, (b) -> b.appendUnscaledDecimalArray(values));
+  }
+
+  /**
+   * Create a new decimal vector from double floats with scale and decimal type.
+   * All doubles will be rescaled according to [[scale]]. Then the integral part of rescaled double will be put to ColumnVector.
+   * If any overflow occurs in extracting integral part, an IllegalArgumentException will be thrown.
+   * Compared with scale of [[java.math.BigDecimal]], the scale here represents the opposite meaning.
+   */
+  public static HostColumnVector decimalFromDoubles(DType.DTypeEnum type, int scale, double... values) {
+    long ratio = (long) Math.pow(10, Math.abs(scale));
+    Buffer buffer;
+    if (type == DType.DTypeEnum.DECIMAL32) {
+      buffer = IntBuffer.allocate(values.length);
```

Comment:
I don't really like the idea of creating a buffer to write the values into, so that we can turn around any copy it.  Why are we not just writing the data directly into `data`?